cmd/gardenlet/app/gardenlet.go

```diff
@@ -418,6 +420,10 @@ func NewGardenlet(ctx context.Context, cfg *config.GardenletConfiguration) (*Gar
 		certificateManager = certificate.NewCertificateManager(clientMap, seedClient.Client(), cfg)
 	}
 
+	if err := client.IgnoreNotFound(seedClient.Client().Delete(ctx, &schedulingv1.PriorityClass{ObjectMeta: metav1.ObjectMeta{Name: "gardenlet"}})); err != nil {
```

Comment:
Put that into `startControllers` or the controller factory, so that this only runs once gardenlet has acquired leadership